qsiprep/interfaces/eddy.py

```diff
@@ -125,6 +130,21 @@ def _run_interface(self, runtime):
         # these have already had HMC, SDC applied
         self._results["forward_transforms"] = []
         self._results["forward_warps"] = []
+
+        # Based on the eddy config, determine whether to send a json argument
+        with open(self.inputs.eddy_config, "r") as eddy_cfg_f:
+            eddy_config = json.load(eddy_cfg_f)
+            # json file is only allowed if mporder is defined
+            if "mporder" in eddy_config:
+                self._results["json_file"] = self.inputs.json_file
+
+        #  Get the multiband factor from the json file and convert it to an arg
+        if isdefined(self.inputs.json_file):
+            with open(self.inputs.json_file) as json_f:
+                input_sidecar = json.load(json_f)
+            if "MultibandAccelerationFactor" in input_sidecar:
+                self._results["multiband_factor"] = input_sidecar["MultibandAccelerationFactor"]
```

Comment:
What does Eddy use `multiband_factor` for?

Reply:
Asking because I worry that multiband factor might be used to infer slice timing. There are some cases (e.g., the CMRR EPI sequence with certain combinations of MB factor and number of slices) where slices won't actually be in the typical order.

Reply:
The only use I can think of for it would be for qsiprep to check that mporder is < MultibandAccelerationFactor

Reply:
It uses it to determine how many groups of slices it should expect. I think it is used internally to help figure out how to divide up the slice timings if it receives a json file instead of its "slspec" format

Reply:
Oh wait, this is going into the JSON file to pass to eddy? Yes I think it needs the factor to know how many groups there are

Reply:
I think it doesn't infer slice timings from the number of groups unless you use the deprecated `--mb` and `--mb_offs` options

Reply:
Yes per the user guide "The current version of eddy allows for a more detailed description of the multi-band structure (see --slspec [below](https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/eddy/UsersGuide#A--slspec)), and if one wants to do slice-to-volume motion correction it is necessary to use that description. It is still possible to use --mb, but it is discouraged and it will be deprecated in the next release."

Reply:
ok, removed this section so mb won't be defined